app/src/main/java/kr/yjkim/book_search/ui/ListViewModel.kt

```diff
@@ -1,18 +1,18 @@
 package kr.yjkim.book_search.ui
 
-import androidx.lifecycle.LiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.ViewModelProvider
 import androidx.lifecycle.viewModelScope
 import androidx.lifecycle.viewmodel.initializer
 import androidx.lifecycle.viewmodel.viewModelFactory
+import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.launch
 import kr.yjkim.book_search.data.BookSearchRepository
-import kr.yjkim.book_search.data.schema.BookItem
+import kr.yjkim.book_search.util.ResultUiState
 
 class ListViewModel(private val repository: BookSearchRepository): ViewModel() {
 
-    val searchResult: LiveData<Result<List<BookItem>>> = repository.searchResult
+    val searchResult: StateFlow<ResultUiState> = repository.searchResult
```

Comment:
요기도 `asStateFlow()` 쓰면 좋을 것 같네요 😄

Reply:
여기는 `MutableStateFlow` 가 아니어서 `asStateFlow()` 가 적용이 안 되는데... 
그럼 `MutableStateFlow` 로 캐스팅한 다음 다시 `asStateFlow()` 를 써야 할까요? 

```suggestion
    val searchResult: StateFlow<ResultUiState> = (repository.searchResult as MutableStateFlow).asStateFlow()
```

Reply:
아! 여긴 searchResult 가 이미 StateFlow 이군요.

그럼 괜찮습니다!